tests/test_local_task_broadcast.py

```diff
@@ -0,0 +1,222 @@
+
+
+@tractor.context
+async def echo_sequences(
+
+    ctx:  tractor.Context,
+
+) -> None:
+    '''Bidir streaming endpoint which will stream
+    back any sequence it is sent item-wise.
+
+    '''
+    await ctx.started()
+
+    async with ctx.open_stream() as stream:
+        async for sequence in stream:
+            seq = list(sequence)
+            for value in seq:
+                print(f'sending {value}')
+                await stream.send(value)
+
+
+async def ensure_sequence(
+    stream: tractor.ReceiveMsgStream,
+    sequence: list,
+) -> None:
+
+    name = current_task().name
+    async with stream.subscribe() as bcaster:
+        assert not isinstance(bcaster, type(stream))
+        async for value in bcaster:
+            print(f'{name} rx: {value}')
+            assert value == sequence[0]
+            sequence.remove(value)
+
+            if not sequence:
+                # fully consumed
+                break
+
+
+def test_stream_fan_out_to_local_subscriptions(
+    arb_addr,
+    start_method,
+):
+
+    sequence = list(range(1000))
+
+    async def main():
+
+        async with tractor.open_nursery(
+            arbiter_addr=arb_addr,
+            start_method=start_method,
+        ) as tn:
+
+            portal = await tn.start_actor(
+                'sequence_echoer',
+                enable_modules=[__name__],
+            )
+
+            async with portal.open_context(
+                echo_sequences,
+            ) as (ctx, first):
+
+                assert first is None
+                async with ctx.open_stream() as stream:
+
+                    async with trio.open_nursery() as n:
+                        for i in range(10):
+                            n.start_soon(
+                                ensure_sequence,
+                                stream,
+                                sequence.copy(),
+                                name=f'consumer_{i}',
+                            )
+
+                        await stream.send(tuple(sequence))
+
+                        async for value in stream:
+                            print(f'source stream rx: {value}')
+                            assert value == sequence[0]
+                            sequence.remove(value)
+
+                            if not sequence:
+                                # fully consumed
+                                break
+
+            await portal.cancel_actor()
+
+    trio.run(main)
+
+
+def test_ensure_slow_consumers_lag_out(
+    arb_addr,
+    start_method,
+):
+    '''This is a pure local task test; no tractor
+    machinery is really required.
+
+    '''
+    async def main():
+
+        # make sure it all works within the runtime
+        async with tractor.open_root_actor():
+
+            num_laggers = 4
+            laggers: dict[str, int] = {}
+            retries = 3
+            size = 100
+            tx, rx = trio.open_memory_channel(size)
+            brx = broadcast_receiver(rx, size)
```

Comment:
I'm almost tempted to enforce this as an async context manager, something I kinda wish `open_memory_channel()` did as well.